<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title id="head-title">blog_report_20250623_152219.html</title>
      <style type="text/css">body {
  font-family: Helvetica, Arial, sans-serif;
  font-size: 12px;
  /* do not increase min-width as some may use split screens */
  min-width: 800px;
  color: #999;
}

h1 {
  font-size: 24px;
  color: black;
}

h2 {
  font-size: 16px;
  color: black;
}

p {
  color: black;
}

a {
  color: #999;
}

table {
  border-collapse: collapse;
}

/******************************
 * SUMMARY INFORMATION
 ******************************/
#environment td {
  padding: 5px;
  border: 1px solid #e6e6e6;
  vertical-align: top;
}
#environment tr:nth-child(odd) {
  background-color: #f6f6f6;
}
#environment ul {
  margin: 0;
  padding: 0 20px;
}

/******************************
 * TEST RESULT COLORS
 ******************************/
span.passed,
.passed .col-result {
  color: green;
}

span.skipped,
span.xfailed,
span.rerun,
.skipped .col-result,
.xfailed .col-result,
.rerun .col-result {
  color: orange;
}

span.error,
span.failed,
span.xpassed,
.error .col-result,
.failed .col-result,
.xpassed .col-result {
  color: red;
}

.col-links__extra {
  margin-right: 3px;
}

/******************************
 * RESULTS TABLE
 *
 * 1. Table Layout
 * 2. Extra
 * 3. Sorting items
 *
 ******************************/
/*------------------
 * 1. Table Layout
 *------------------*/
#results-table {
  border: 1px solid #e6e6e6;
  color: #999;
  font-size: 12px;
  width: 100%;
}
#results-table th,
#results-table td {
  padding: 5px;
  border: 1px solid #e6e6e6;
  text-align: left;
}
#results-table th {
  font-weight: bold;
}

/*------------------
 * 2. Extra
 *------------------*/
.logwrapper {
  max-height: 230px;
  overflow-y: scroll;
  background-color: #e6e6e6;
}
.logwrapper.expanded {
  max-height: none;
}
.logwrapper.expanded .logexpander:after {
  content: "collapse [-]";
}
.logwrapper .logexpander {
  z-index: 1;
  position: sticky;
  top: 10px;
  width: max-content;
  border: 1px solid;
  border-radius: 3px;
  padding: 5px 7px;
  margin: 10px 0 10px calc(100% - 80px);
  cursor: pointer;
  background-color: #e6e6e6;
}
.logwrapper .logexpander:after {
  content: "expand [+]";
}
.logwrapper .logexpander:hover {
  color: #000;
  border-color: #000;
}
.logwrapper .log {
  min-height: 40px;
  position: relative;
  top: -50px;
  height: calc(100% + 50px);
  border: 1px solid #e6e6e6;
  color: black;
  display: block;
  font-family: "Courier New", Courier, monospace;
  padding: 5px;
  padding-right: 80px;
  white-space: pre-wrap;
}

div.media {
  border: 1px solid #e6e6e6;
  float: right;
  height: 240px;
  margin: 0 5px;
  overflow: hidden;
  width: 320px;
}

.media-container {
  display: grid;
  grid-template-columns: 25px auto 25px;
  align-items: center;
  flex: 1 1;
  overflow: hidden;
  height: 200px;
}

.media-container--fullscreen {
  grid-template-columns: 0px auto 0px;
}

.media-container__nav--right,
.media-container__nav--left {
  text-align: center;
  cursor: pointer;
}

.media-container__viewport {
  cursor: pointer;
  text-align: center;
  height: inherit;
}
.media-container__viewport img,
.media-container__viewport video {
  object-fit: cover;
  width: 100%;
  max-height: 100%;
}

.media__name,
.media__counter {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  flex: 0 0 25px;
  align-items: center;
}

.collapsible td:not(.col-links) {
  cursor: pointer;
}
.collapsible td:not(.col-links):hover::after {
  color: #bbb;
  font-style: italic;
  cursor: pointer;
}

.col-result {
  width: 130px;
}
.col-result:hover::after {
  content: " (hide details)";
}

.col-result.collapsed:hover::after {
  content: " (show details)";
}

#environment-header h2:hover::after {
  content: " (hide details)";
  color: #bbb;
  font-style: italic;
  cursor: pointer;
  font-size: 12px;
}

#environment-header.collapsed h2:hover::after {
  content: " (show details)";
  color: #bbb;
  font-style: italic;
  cursor: pointer;
  font-size: 12px;
}

/*------------------
 * 3. Sorting items
 *------------------*/
.sortable {
  cursor: pointer;
}
.sortable.desc:after {
  content: " ";
  position: relative;
  left: 5px;
  bottom: -12.5px;
  border: 10px solid #4caf50;
  border-bottom: 0;
  border-left-color: transparent;
  border-right-color: transparent;
}
.sortable.asc:after {
  content: " ";
  position: relative;
  left: 5px;
  bottom: 12.5px;
  border: 10px solid #4caf50;
  border-top: 0;
  border-left-color: transparent;
  border-right-color: transparent;
}

.hidden, .summary__reload__button.hidden {
  display: none;
}

.summary__data {
  flex: 0 0 550px;
}
.summary__reload {
  flex: 1 1;
  display: flex;
  justify-content: center;
}
.summary__reload__button {
  flex: 0 0 300px;
  display: flex;
  color: white;
  font-weight: bold;
  background-color: #4caf50;
  text-align: center;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
  cursor: pointer;
}
.summary__reload__button:hover {
  background-color: #46a049;
}
.summary__spacer {
  flex: 0 0 550px;
}

.controls {
  display: flex;
  justify-content: space-between;
}

.filters,
.collapse {
  display: flex;
  align-items: center;
}
.filters button,
.collapse button {
  color: #999;
  border: none;
  background: none;
  cursor: pointer;
  text-decoration: underline;
}
.filters button:hover,
.collapse button:hover {
  color: #ccc;
}

.filter__label {
  margin-right: 10px;
}

      </style>
    
  </head>
  <body>
    <h1 id="title">blog_report_20250623_152219.html</h1>
    <p>Report generated on 23-Jun-2025 at 15:32:18 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a>
        v4.1.1</p>
    <div id="environment-header">
      <h2>Environment</h2>
    </div>
    <table id="environment"></table>
    <!-- TEMPLATES -->
      <template id="template_environment_row">
      <tr>
        <td></td>
        <td></td>
      </tr>
    </template>
    <template id="template_results-table__body--empty">
      <tbody class="results-table-row">
        <tr id="not-found-message">
          <td colspan="4">No results found. Check the filters.</th>
        </tr>
    </template>
    <template id="template_results-table__tbody">
      <tbody class="results-table-row">
        <tr class="collapsible">
        </tr>
        <tr class="extras-row">
          <td class="extra" colspan="4">
            <div class="extraHTML"></div>
            <div class="media">
              <div class="media-container">
                  <div class="media-container__nav--left"><</div>
                  <div class="media-container__viewport">
                    <img src="" />
                    <video controls>
                      <source src="" type="video/mp4">
                    </video>
                  </div>
                  <div class="media-container__nav--right">></div>
                </div>
                <div class="media__name"></div>
                <div class="media__counter"></div>
            </div>
            <div class="logwrapper">
              <div class="logexpander"></div>
              <div class="log"></div>
            </div>
          </td>
        </tr>
      </tbody>
    </template>
    <!-- END TEMPLATES -->
    <div class="summary">
      <div class="summary__data">
        <h2>Summary</h2>
        <div class="additional-summary prefix">
        </div>
        <p class="run-count">18 tests took 00:09:53.</p>
        <p class="filter">(Un)check the boxes to filter the results.</p>
        <div class="summary__reload">
          <div class="summary__reload__button hidden" onclick="location.reload()">
            <div>There are still tests running. <br />Reload this page to get the latest results!</div>
          </div>
        </div>
        <div class="summary__spacer"></div>
        <div class="controls">
          <div class="filters">
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="failed" />
            <span class="failed">11 Failed,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="passed" />
            <span class="passed">7 Passed,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="skipped" />
            <span class="skipped">3 Skipped,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="xfailed" disabled/>
            <span class="xfailed">0 Expected failures,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="xpassed" disabled/>
            <span class="xpassed">0 Unexpected passes,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="error" disabled/>
            <span class="error">0 Errors,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="rerun" disabled/>
            <span class="rerun">0 Reruns</span>
          </div>
          <div class="collapse">
            <button id="show_all_details">Show all details</button>&nbsp;/&nbsp;<button id="hide_all_details">Hide all details</button>
          </div>
        </div>
      </div>
      <div class="additional-summary summary">
      </div>
      <div class="additional-summary postfix">
      </div>
    </div>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable" data-column-type="result">Result</th>
          <th class="sortable" data-column-type="testId">Test</th>
          <th class="sortable" data-column-type="duration">Duration</th>
          <th>Links</th>
        </tr>
      </thead>
    </table>
  </body>
  <footer>
    <div id="data-container" data-jsonblob="{&#34;environment&#34;: {&#34;Python&#34;: &#34;3.12.10&#34;, &#34;Platform&#34;: &#34;Windows-11-10.0.26100-SP0&#34;, &#34;Packages&#34;: {&#34;pytest&#34;: &#34;8.4.1&#34;, &#34;pluggy&#34;: &#34;1.6.0&#34;}, &#34;Plugins&#34;: {&#34;allure-pytest&#34;: &#34;2.14.3&#34;, &#34;Faker&#34;: &#34;37.4.0&#34;, &#34;html&#34;: &#34;4.1.1&#34;, &#34;metadata&#34;: &#34;3.1.1&#34;}}, &#34;tests&#34;: {&#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_can_read_blogs&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_can_read_blogs&#34;, &#34;duration&#34;: &#34;00:00:59&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_can_read_blogs&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:59&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:163: in test_guest_can_read_blogs\n    assert self.blog_page.navigate_to_blog_list(), &amp;quot;Failed to navigate to blog list&amp;quot;\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\npages\\blog_page_new.py:91: in navigate_to_blog_list\n    self.driver.get(blog_url)\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:472: in get\n    self.execute(Command.GET, {&amp;quot;url&amp;quot;: url})\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:447: in execute\n    self.error_handler.check_response(response)\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:232: in check_response\n    raise exception_class(message, screen, stacktrace)\nE   selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE     (Session info: MicrosoftEdge=137.0.3296.93)\nE   Stacktrace:\nE   \tGetHandleVerifier [0x0x7ff7538a3985+25605]\nE   \t(No symbol) [0x0x7ff7537f3a90]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\nE   \t(No symbol) [0x0x7ff7535b9110]\nE   \t(No symbol) [0x0x7ff7535ad3af]\nE   \t(No symbol) [0x0x7ff7535aea26]\nE   \t(No symbol) [0x0x7ff7535ad619]\nE   \t(No symbol) [0x0x7ff7535ad21c]\nE   \t(No symbol) [0x0x7ff7535acf68]\nE   \t(No symbol) [0x0x7ff7535aaf5b]\nE   \t(No symbol) [0x0x7ff7535ab53c]\nE   \t(No symbol) [0x0x7ff7535bf6ba]\nE   \t(No symbol) [0x0x7ff753647b6e]\nE   \t(No symbol) [0x0x7ff753629e4a]\nE   \t(No symbol) [0x0x7ff7535fea5d]\nE   \t(No symbol) [0x0x7ff7536475df]\nE   \t(No symbol) [0x0x7ff753629bb3]\nE   \t(No symbol) [0x0x7ff7535fdf56]\nE   \t(No symbol) [0x0x7ff7535fd463]\nE   \t(No symbol) [0x0x7ff7535fdd83]\nE   \t(No symbol) [0x0x7ff7536fe0fd]\nE   \t(No symbol) [0x0x7ff75370dd08]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\nE   \t(No symbol) [0x0x7ff753802231]\nE   \t(No symbol) [0x0x7ff7537fa6d4]\nE   \t(No symbol) [0x0x7ff7537fa823]\nE   \t(No symbol) [0x0x7ff7537eb9f6]\nE   \tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\nE   \tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n---------------------------- Captured stdout setup -----------------------------\n\n\ud83d\ude80 Starting GenCare Test Suite - 2025-06-23 15:22:20.915521\n\ud83d\udccd Base URL: http://localhost:5173\n\ud83c\udf10 Browser: edge\n\ud83d\udcf1 Headless: False\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - === Working Blog CRUD Test Suite Started ===\nINFO - \n\ud83e\uddea Starting test: test_guest_can_read_blogs\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2911CBF0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925A5A0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925A840&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925AF30&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925B200&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925B4A0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BB60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BE30&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29698140&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_guest_can_read_blogs\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2911CBF0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925A5A0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925A840&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925AF30&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925B200&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925B4A0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BB60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BE30&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29698140&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/04b772f7a0bb44d6ab464fa65c4cee4d/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_guest_can_read_blogs&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_read_blogs&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_read_blogs&#34;, &#34;duration&#34;: &#34;00:00:60&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_read_blogs&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:60&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:184: in test_admin_can_read_blogs\n    assert self.manual_login_admin(), &amp;quot;Admin login failed&amp;quot;\nE   AssertionError: Admin login failed\nE   assert False\nE    +  where False = manual_login_admin()\nE    +    where manual_login_admin = &amp;lt;test_blog_crud_working.TestBlogCRUDWorking object at 0x0000018F2914BA70&amp;gt;.manual_login_admin\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_admin_can_read_blogs\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_admin_can_read_blogs\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - \ud83d\udd11 Performing manual admin login\nINFO - Navigating to: http://localhost:5173/\nERROR - Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n\n------------------------------ Captured log call -------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:48 \ud83d\udd11 Performing manual admin login\nINFO     BasePage:base_page.py:66 Navigating to: http://localhost:5173/\nERROR    TestBlogCRUDWorking:test_blog_crud_working.py:104 Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296997C0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699910&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699520&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A150&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A3C0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A630&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969AC60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969AF00&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B1D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_admin_can_read_blogs\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296997C0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699910&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699520&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A150&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A3C0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A630&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969AC60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969AF00&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B1D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/d700bf283beafcbab5d797e8ddac5ce4/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_admin_can_read_blogs&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_access_blog_creation&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_access_blog_creation&#34;, &#34;duration&#34;: &#34;00:00:59&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_access_blog_creation&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:59&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:209: in test_admin_can_access_blog_creation\n    assert self.manual_login_admin(), &amp;quot;Admin login failed&amp;quot;\nE   AssertionError: Admin login failed\nE   assert False\nE    +  where False = manual_login_admin()\nE    +    where manual_login_admin = &amp;lt;test_blog_crud_working.TestBlogCRUDWorking object at 0x0000018F2914BB90&amp;gt;.manual_login_admin\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_admin_can_access_blog_creation\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_admin_can_access_blog_creation\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - \ud83d\udd11 Performing manual admin login\nINFO - Navigating to: http://localhost:5173/\nERROR - Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n\n------------------------------ Captured log call -------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:48 \ud83d\udd11 Performing manual admin login\nINFO     BasePage:base_page.py:66 Navigating to: http://localhost:5173/\nERROR    TestBlogCRUDWorking:test_blog_crud_working.py:104 Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BE60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29259220&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2911E870&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B950&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B710&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A3F0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29698B90&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969BDA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969AC00&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_admin_can_access_blog_creation\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BE60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29259220&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2911E870&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B950&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B710&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A3F0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29698B90&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969BDA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969AC00&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/51d47eafb9fd7b32600d590c65620d48/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_admin_can_access_blog_creation&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_cannot_access_blog_creation&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_cannot_access_blog_creation&#34;, &#34;duration&#34;: &#34;00:00:59&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_cannot_access_blog_creation&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:59&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:243: in test_guest_cannot_access_blog_creation\n    assert self.blog_page.navigate_to_blog_list(), &amp;quot;Failed to navigate to blog list&amp;quot;\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\npages\\blog_page_new.py:91: in navigate_to_blog_list\n    self.driver.get(blog_url)\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:472: in get\n    self.execute(Command.GET, {&amp;quot;url&amp;quot;: url})\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:447: in execute\n    self.error_handler.check_response(response)\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:232: in check_response\n    raise exception_class(message, screen, stacktrace)\nE   selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE     (Session info: MicrosoftEdge=137.0.3296.93)\nE   Stacktrace:\nE   \tGetHandleVerifier [0x0x7ff7538a3985+25605]\nE   \t(No symbol) [0x0x7ff7537f3a90]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\nE   \t(No symbol) [0x0x7ff7535b9110]\nE   \t(No symbol) [0x0x7ff7535ad3af]\nE   \t(No symbol) [0x0x7ff7535aea26]\nE   \t(No symbol) [0x0x7ff7535ad619]\nE   \t(No symbol) [0x0x7ff7535ad21c]\nE   \t(No symbol) [0x0x7ff7535acf68]\nE   \t(No symbol) [0x0x7ff7535aaf5b]\nE   \t(No symbol) [0x0x7ff7535ab53c]\nE   \t(No symbol) [0x0x7ff7535bf6ba]\nE   \t(No symbol) [0x0x7ff753647b6e]\nE   \t(No symbol) [0x0x7ff753629e4a]\nE   \t(No symbol) [0x0x7ff7535fea5d]\nE   \t(No symbol) [0x0x7ff7536475df]\nE   \t(No symbol) [0x0x7ff753629bb3]\nE   \t(No symbol) [0x0x7ff7535fdf56]\nE   \t(No symbol) [0x0x7ff7535fd463]\nE   \t(No symbol) [0x0x7ff7535fdd83]\nE   \t(No symbol) [0x0x7ff7536fe0fd]\nE   \t(No symbol) [0x0x7ff75370dd08]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\nE   \t(No symbol) [0x0x7ff753802231]\nE   \t(No symbol) [0x0x7ff7537fa6d4]\nE   \t(No symbol) [0x0x7ff7537fa823]\nE   \t(No symbol) [0x0x7ff7537eb9f6]\nE   \tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\nE   \tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_guest_cannot_access_blog_creation\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_guest_cannot_access_blog_creation\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B170&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B620&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699B20&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B380&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969BDD0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A210&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B680&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699D30&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699AC0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_guest_cannot_access_blog_creation\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B170&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B620&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699B20&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B380&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969BDD0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A210&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B680&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699D30&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699AC0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/ab937d3c4e406eb391f5ab2434cfc0ae/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_guest_cannot_access_blog_creation&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_see_comment_functionality&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_see_comment_functionality&#34;, &#34;duration&#34;: &#34;00:00:58&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_can_see_comment_functionality&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:58&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:286: in test_admin_can_see_comment_functionality\n    assert self.manual_login_admin(), &amp;quot;Admin login failed&amp;quot;\nE   AssertionError: Admin login failed\nE   assert False\nE    +  where False = manual_login_admin()\nE    +    where manual_login_admin = &amp;lt;test_blog_crud_working.TestBlogCRUDWorking object at 0x0000018F29198050&amp;gt;.manual_login_admin\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_admin_can_see_comment_functionality\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_admin_can_see_comment_functionality\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - \ud83d\udd11 Performing manual admin login\nINFO - Navigating to: http://localhost:5173/\nERROR - Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n\n------------------------------ Captured log call -------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:48 \ud83d\udd11 Performing manual admin login\nINFO     BasePage:base_page.py:66 Navigating to: http://localhost:5173/\nERROR    TestBlogCRUDWorking:test_blog_crud_working.py:104 Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296993A0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296990D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699C40&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B620&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296987D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DEB0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C290&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C950&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969CB60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_admin_can_see_comment_functionality\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296993A0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296990D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F29699C40&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969B620&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F296987D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DEB0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C290&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C950&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969CB60&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/38319b44e8084964b41561f8d2b46d49/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_admin_can_see_comment_functionality&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_cannot_comment&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_cannot_comment&#34;, &#34;duration&#34;: &#34;00:01:04&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_guest_cannot_comment&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:01:04&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:325: in test_guest_cannot_comment\n    assert self.blog_page.navigate_to_blog_list(), &amp;quot;Failed to navigate to blog list&amp;quot;\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\npages\\blog_page_new.py:91: in navigate_to_blog_list\n    self.driver.get(blog_url)\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:472: in get\n    self.execute(Command.GET, {&amp;quot;url&amp;quot;: url})\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:447: in execute\n    self.error_handler.check_response(response)\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:232: in check_response\n    raise exception_class(message, screen, stacktrace)\nE   selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE     (Session info: MicrosoftEdge=137.0.3296.93)\nE   Stacktrace:\nE   \tGetHandleVerifier [0x0x7ff7538a3985+25605]\nE   \t(No symbol) [0x0x7ff7537f3a90]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\nE   \t(No symbol) [0x0x7ff7535b9110]\nE   \t(No symbol) [0x0x7ff7535ad3af]\nE   \t(No symbol) [0x0x7ff7535aea26]\nE   \t(No symbol) [0x0x7ff7535ad619]\nE   \t(No symbol) [0x0x7ff7535ad21c]\nE   \t(No symbol) [0x0x7ff7535acf68]\nE   \t(No symbol) [0x0x7ff7535aaf5b]\nE   \t(No symbol) [0x0x7ff7535ab53c]\nE   \t(No symbol) [0x0x7ff7535bf6ba]\nE   \t(No symbol) [0x0x7ff753647b6e]\nE   \t(No symbol) [0x0x7ff753629e4a]\nE   \t(No symbol) [0x0x7ff7535fea5d]\nE   \t(No symbol) [0x0x7ff7536475df]\nE   \t(No symbol) [0x0x7ff753629bb3]\nE   \t(No symbol) [0x0x7ff7535fdf56]\nE   \t(No symbol) [0x0x7ff7535fd463]\nE   \t(No symbol) [0x0x7ff7535fdd83]\nE   \t(No symbol) [0x0x7ff7536fe0fd]\nE   \t(No symbol) [0x0x7ff75370dd08]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\nE   \t(No symbol) [0x0x7ff753802231]\nE   \t(No symbol) [0x0x7ff7537fa6d4]\nE   \t(No symbol) [0x0x7ff7537fa823]\nE   \t(No symbol) [0x0x7ff7537eb9f6]\nE   \tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\nE   \tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_guest_cannot_comment\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_guest_cannot_comment\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969ABA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A510&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969ACC0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925ABA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925B1D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BDD0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D430&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969EBA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969E750&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_guest_cannot_comment\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969ABA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969A510&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969ACC0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925ABA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925B1D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2925BDD0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D430&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969EBA0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969E750&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/dd131821b388e15ab8e8b679107d2241/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_guest_cannot_comment&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_blog_navigation_flow&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_blog_navigation_flow&#34;, &#34;duration&#34;: &#34;00:00:58&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_blog_navigation_flow&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:58&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:369: in test_blog_navigation_flow\n    assert self.manual_login_admin(), &amp;quot;Admin login failed&amp;quot;\nE   AssertionError: Admin login failed\nE   assert False\nE    +  where False = manual_login_admin()\nE    +    where manual_login_admin = &amp;lt;test_blog_crud_working.TestBlogCRUDWorking object at 0x0000018F28935520&amp;gt;.manual_login_admin\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_blog_navigation_flow\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_blog_navigation_flow\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - \ud83d\udd11 Performing manual admin login\nINFO - Navigating to: http://localhost:5173/\nERROR - Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n\n------------------------------ Captured log call -------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:48 \ud83d\udd11 Performing manual admin login\nINFO     BasePage:base_page.py:66 Navigating to: http://localhost:5173/\nERROR    TestBlogCRUDWorking:test_blog_crud_working.py:104 Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DB80&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969F050&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D3D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C650&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C290&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DD00&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C2F0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D040&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C3B0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_blog_navigation_flow\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DB80&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969F050&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D3D0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C650&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C290&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DD00&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C2F0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D040&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C3B0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/34837d3334306e43b38f60180ca7e69e/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_blog_navigation_flow&#34;}], &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_full_blog_access&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_full_blog_access&#34;, &#34;duration&#34;: &#34;00:00:58&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_crud_working.py::TestBlogCRUDWorking::test_admin_full_blog_access&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:58&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_crud_working.py:411: in test_admin_full_blog_access\n    assert self.manual_login_admin(), &amp;quot;Admin login failed&amp;quot;\nE   AssertionError: Admin login failed\nE   assert False\nE    +  where False = manual_login_admin()\nE    +    where manual_login_admin = &amp;lt;test_blog_crud_working.TestBlogCRUDWorking object at 0x0000018F29198170&amp;gt;.manual_login_admin\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - \n\ud83e\uddea Starting test: test_admin_full_blog_access\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:27 \n\ud83e\uddea Starting test: test_admin_full_blog_access\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - \ud83d\udd11 Performing manual admin login\nINFO - Navigating to: http://localhost:5173/\nERROR - Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n\n------------------------------ Captured log call -------------------------------\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:48 \ud83d\udd11 Performing manual admin login\nINFO     BasePage:base_page.py:66 Navigating to: http://localhost:5173/\nERROR    TestBlogCRUDWorking:test_blog_crud_working.py:104 Manual login failed: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: MicrosoftEdge=137.0.3296.93)\nStacktrace:\n\tGetHandleVerifier [0x0x7ff7538a3985+25605]\n\t(No symbol) [0x0x7ff7537f3a90]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\n\t(No symbol) [0x0x7ff7535b9110]\n\t(No symbol) [0x0x7ff7535ad3af]\n\t(No symbol) [0x0x7ff7535aea26]\n\t(No symbol) [0x0x7ff7535ad619]\n\t(No symbol) [0x0x7ff7535ad21c]\n\t(No symbol) [0x0x7ff7535acf68]\n\t(No symbol) [0x0x7ff7535aaf5b]\n\t(No symbol) [0x0x7ff7535ab53c]\n\t(No symbol) [0x0x7ff7535bf6ba]\n\t(No symbol) [0x0x7ff753647b6e]\n\t(No symbol) [0x0x7ff753629e4a]\n\t(No symbol) [0x0x7ff7535fea5d]\n\t(No symbol) [0x0x7ff7536475df]\n\t(No symbol) [0x0x7ff753629bb3]\n\t(No symbol) [0x0x7ff7535fdf56]\n\t(No symbol) [0x0x7ff7535fd463]\n\t(No symbol) [0x0x7ff7535fdd83]\n\t(No symbol) [0x0x7ff7536fe0fd]\n\t(No symbol) [0x0x7ff75370dd08]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\n\tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\n\t(No symbol) [0x0x7ff753802231]\n\t(No symbol) [0x0x7ff7537fa6d4]\n\t(No symbol) [0x0x7ff7537fa823]\n\t(No symbol) [0x0x7ff7537eb9f6]\n\tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\n\tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C2F0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C860&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969E900&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969E570&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969CFB0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C950&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D7C0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DB80&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING - Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969EE70&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nINFO - No logout button found - user not logged in\nINFO - \u2705 Test completed: test_admin_full_blog_access\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C2F0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C860&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969E900&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969E570&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969CFB0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969C950&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969D7C0&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969DB80&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NewConnectionError(&amp;#x27;&amp;lt;urllib3.connection.HTTPConnection object at 0x0000018F2969EE70&amp;gt;: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it&amp;#x27;)&amp;#x27;: /session/a2a7d0736f45c1de3125ad98a1d942e4/element\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:127 No logout button found - user not logged in\nINFO     TestBlogCRUDWorking:test_blog_crud_working.py:35 \u2705 Test completed: test_admin_full_blog_access&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_list_page_loads&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_list_page_loads&#34;, &#34;duration&#34;: &#34;00:00:08&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_list_page_loads&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:08&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_navigation_menu&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_navigation_menu&#34;, &#34;duration&#34;: &#34;00:00:12&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_navigation_menu&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:12&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;pages\\blog_page_new.py:117: in click_blog_menu\n    blog_link = self.wait_for_element_clickable(self.BLOG_MENU_LINK, timeout=5)\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\npages\\base_page.py:98: in wait_for_element_clickable\n    return self.wait_helpers.wait_for_element_clickable(locator, timeout)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nutils\\wait_helpers.py:51: in wait_for_element_clickable\n    element = wait.until(EC.element_to_be_clickable(locator))\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nC:\\Users\\Dell\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:138: in until\n    raise TimeoutException(message, screen, stacktrace)\nE   selenium.common.exceptions.TimeoutException: Message: \nE   Stacktrace:\nE   \tGetHandleVerifier [0x0x7ff7538a3985+25605]\nE   \t(No symbol) [0x0x7ff7537f3a90]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753b26fca+1962554]\nE   \t(No symbol) [0x0x7ff7536086c4]\nE   \t(No symbol) [0x0x7ff75360898b]\nE   \t(No symbol) [0x0x7ff753649af7]\nE   \t(No symbol) [0x0x7ff753629e8f]\nE   \t(No symbol) [0x0x7ff7535fea5d]\nE   \t(No symbol) [0x0x7ff7536475df]\nE   \t(No symbol) [0x0x7ff753629bb3]\nE   \t(No symbol) [0x0x7ff7535fdf56]\nE   \t(No symbol) [0x0x7ff7535fd463]\nE   \t(No symbol) [0x0x7ff7535fdd83]\nE   \t(No symbol) [0x0x7ff7536fe0fd]\nE   \t(No symbol) [0x0x7ff75370dd08]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753988959+265161]\nE   \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff753990261+296145]\nE   \t(No symbol) [0x0x7ff753802231]\nE   \t(No symbol) [0x0x7ff7537fa6d4]\nE   \t(No symbol) [0x0x7ff7537fa823]\nE   \t(No symbol) [0x0x7ff7537eb9f6]\nE   \tBaseThreadInitThunk [0x0x7fffdeafe8d7+23]\nE   \tRtlUserThreadStart [0x0x7fffe025c34c+44]\n\nDuring handling of the above exception, another exception occurred:\ntests\\test_blog_new.py:69: in test_blog_navigation_menu\n    clicked = self.blog_page.click_blog_menu()\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\npages\\blog_page_new.py:124: in click_blog_menu\n    blog_nav = self.wait_for_element(self.BLOG_NAV_ITEM, timeout=3)\n               ^^^^^^^^^^^^^^^^^^^^^\nE   AttributeError: &amp;#x27;BlogPageNew&amp;#x27; object has no attribute &amp;#x27;wait_for_element&amp;#x27;\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to home: http://localhost:5173\nERROR - Element not clickable within 5s: (&amp;#x27;link text&amp;#x27;, &amp;#x27;Blogs&amp;#x27;)\nERROR - Element not clickable: (&amp;#x27;link text&amp;#x27;, &amp;#x27;Blogs&amp;#x27;)\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:108 Navigating to home: http://localhost:5173\nERROR    utils.wait_helpers:wait_helpers.py:55 Element not clickable within 5s: (&amp;#x27;link text&amp;#x27;, &amp;#x27;Blogs&amp;#x27;)\nERROR    BlogPageNew:base_page.py:100 Element not clickable: (&amp;#x27;link text&amp;#x27;, &amp;#x27;Blogs&amp;#x27;)\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_listing_display&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_listing_display&#34;, &#34;duration&#34;: &#34;00:00:09&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_listing_display&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:09&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_new.py:93: in test_blog_listing_display\n    assert first_title is not None, &amp;quot;Should be able to get first blog title&amp;quot;\nE   AssertionError: Should be able to get first blog title\nE   assert None is not None\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\nINFO - Found 10 blog cards on page\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\nINFO     BlogPageNew:blog_page_new.py:201 Found 10 blog cards on page\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_guest_blog_creation_access&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_guest_blog_creation_access&#34;, &#34;duration&#34;: &#34;00:00:16&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_guest_blog_creation_access&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:16&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to home: http://localhost:5173\nINFO - Navigating to blog create: http://localhost:5173/blogs/create\nERROR - Element not visible within 3s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR - Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR - Element not visible within 2s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;.access-denied, .unauthorized, [data-testid=&amp;#x27;access-denied&amp;#x27;]&amp;quot;)\nERROR - Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;.access-denied, .unauthorized, [data-testid=&amp;#x27;access-denied&amp;#x27;]&amp;quot;)\nERROR - Element not visible within 2s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;button[data-testid=&amp;#x27;create-blog&amp;#x27;], a[href*=&amp;#x27;create&amp;#x27;]&amp;quot;)\nERROR - Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;button[data-testid=&amp;#x27;create-blog&amp;#x27;], a[href*=&amp;#x27;create&amp;#x27;]&amp;quot;)\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:108 Navigating to home: http://localhost:5173\nINFO     BlogPageNew:blog_page_new.py:99 Navigating to blog create: http://localhost:5173/blogs/create\nERROR    utils.wait_helpers:wait_helpers.py:80 Element not visible within 3s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR    BlogPageNew:base_page.py:117 Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR    utils.wait_helpers:wait_helpers.py:80 Element not visible within 2s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;.access-denied, .unauthorized, [data-testid=&amp;#x27;access-denied&amp;#x27;]&amp;quot;)\nERROR    BlogPageNew:base_page.py:117 Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;.access-denied, .unauthorized, [data-testid=&amp;#x27;access-denied&amp;#x27;]&amp;quot;)\nERROR    utils.wait_helpers:wait_helpers.py:80 Element not visible within 2s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;button[data-testid=&amp;#x27;create-blog&amp;#x27;], a[href*=&amp;#x27;create&amp;#x27;]&amp;quot;)\nERROR    BlogPageNew:base_page.py:117 Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;button[data-testid=&amp;#x27;create-blog&amp;#x27;], a[href*=&amp;#x27;create&amp;#x27;]&amp;quot;)\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_guest_blog_listing_access&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_guest_blog_listing_access&#34;, &#34;duration&#34;: &#34;00:00:07&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_guest_blog_listing_access&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:07&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to home: http://localhost:5173\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:108 Navigating to home: http://localhost:5173\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_creation_form_validation&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Skipped&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_creation_form_validation&#34;, &#34;duration&#34;: &#34;00:00:09&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Skipped&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_creation_form_validation&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:09&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;(&amp;#x27;D:\\\\Ky5\\\\SWP\\\\Group3SWP\\\\selenium-tests\\\\tests\\\\test_blog_new.py&amp;#x27;, 160, &amp;#x27;Skipped: Blog creation not accessible - requires authentication&amp;#x27;)\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog create: http://localhost:5173/blogs/create\nERROR - Element not visible within 3s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR - Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:99 Navigating to blog create: http://localhost:5173/blogs/create\nERROR    utils.wait_helpers:wait_helpers.py:80 Element not visible within 3s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR    BlogPageNew:base_page.py:117 Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_creation_workflow&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Skipped&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_creation_workflow&#34;, &#34;duration&#34;: &#34;00:00:09&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Skipped&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_creation_workflow&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:09&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;(&amp;#x27;D:\\\\Ky5\\\\SWP\\\\Group3SWP\\\\selenium-tests\\\\tests\\\\test_blog_new.py&amp;#x27;, 190, &amp;#x27;Skipped: Blog creation not accessible - requires authentication&amp;#x27;)\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog create: http://localhost:5173/blogs/create\nERROR - Element not visible within 3s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR - Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:99 Navigating to blog create: http://localhost:5173/blogs/create\nERROR    utils.wait_helpers:wait_helpers.py:80 Element not visible within 3s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\nERROR    BlogPageNew:base_page.py:117 Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;form[data-testid=&amp;#x27;blog-form&amp;#x27;], .blog-form&amp;quot;)\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_search&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Skipped&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_search&#34;, &#34;duration&#34;: &#34;00:00:12&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Skipped&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_search&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:12&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;(&amp;#x27;D:\\\\Ky5\\\\SWP\\\\Group3SWP\\\\selenium-tests\\\\tests\\\\test_blog_new.py&amp;#x27;, 243, &amp;#x27;Skipped: Search functionality not available&amp;#x27;)\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\nERROR - Element not visible within 5s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;input[placeholder*=&amp;#x27;Search&amp;#x27;], input[type=&amp;#x27;search&amp;#x27;]&amp;quot;)\nERROR - Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;input[placeholder*=&amp;#x27;Search&amp;#x27;], input[type=&amp;#x27;search&amp;#x27;]&amp;quot;)\nWARNING - Search functionality not found\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\nERROR    utils.wait_helpers:wait_helpers.py:80 Element not visible within 5s: (&amp;#x27;css selector&amp;#x27;, &amp;quot;input[placeholder*=&amp;#x27;Search&amp;#x27;], input[type=&amp;#x27;search&amp;#x27;]&amp;quot;)\nERROR    BlogPageNew:base_page.py:117 Element not visible: (&amp;#x27;css selector&amp;#x27;, &amp;quot;input[placeholder*=&amp;#x27;Search&amp;#x27;], input[type=&amp;#x27;search&amp;#x27;]&amp;quot;)\nWARNING  BlogPageNew:blog_page_new.py:379 Search functionality not found\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_detail_navigation&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_detail_navigation&#34;, &#34;duration&#34;: &#34;00:00:09&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_detail_navigation&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:09&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\nINFO - Found 10 blog cards on page\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\nINFO     BlogPageNew:blog_page_new.py:201 Found 10 blog cards on page\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_page_responsive&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_blog_page_responsive&#34;, &#34;duration&#34;: &#34;00:00:09&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_blog_page_responsive&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:09&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogSystemNew::test_invalid_blog_url&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogSystemNew::test_invalid_blog_url&#34;, &#34;duration&#34;: &#34;00:00:07&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogSystemNew::test_invalid_blog_url&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:07&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestAuthenticatedBlogFeatures::test_authenticated_blog_creation&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Failed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestAuthenticatedBlogFeatures::test_authenticated_blog_creation&#34;, &#34;duration&#34;: &#34;00:00:05&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Failed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestAuthenticatedBlogFeatures::test_authenticated_blog_creation&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:05&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;tests\\test_blog_new.py:362: in test_authenticated_blog_creation\n    login_success = self.login_page.login(&amp;quot;test@gencare.com&amp;quot;, &amp;quot;password123&amp;quot;)\n                    ^^^^^^^^^^^^^^^^^^^^^\nE   AttributeError: &amp;#x27;LoginPage&amp;#x27; object has no attribute &amp;#x27;login&amp;#x27;\n\n---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to: http://localhost:5173/\nINFO - Navigated to home page successfully\nINFO - Navigating to: http://localhost:5173/\nINFO - Clicked element: (&amp;#x27;xpath&amp;#x27;, &amp;quot;//button[contains(text(), &amp;#x27;\u0110\u0103ng nh\u1eadp&amp;#x27;)]&amp;quot;)\nINFO - Login modal opened successfully\n\n------------------------------ Captured log call -------------------------------\nINFO     LoginPage:base_page.py:66 Navigating to: http://localhost:5173/\nINFO     LoginPage:login_page.py:427 Navigated to home page successfully\nINFO     LoginPage:base_page.py:66 Navigating to: http://localhost:5173/\nINFO     LoginPage:base_page.py:134 Clicked element: (&amp;#x27;xpath&amp;#x27;, &amp;quot;//button[contains(text(), &amp;#x27;\u0110\u0103ng nh\u1eadp&amp;#x27;)]&amp;quot;)\nINFO     LoginPage:login_page.py:75 Login modal opened successfully\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}], &#34;tests/test_blog_new.py::TestBlogPerformance::test_blog_listing_load_time&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;tests/test_blog_new.py::TestBlogPerformance::test_blog_listing_load_time&#34;, &#34;duration&#34;: &#34;00:00:06&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/test_blog_new.py::TestBlogPerformance::test_blog_listing_load_time&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:06&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;---------------------------- Captured stderr setup -----------------------------\nINFO - ====== WebDriver manager ======\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Get LATEST edgedriver version for Edge 137.0.3296\nINFO - Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO - Edge WebDriver created successfully\nINFO - Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n------------------------------ Captured log setup ------------------------------\nINFO     WDM:logger.py:11 ====== WebDriver manager ======\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Get LATEST edgedriver version for Edge 137.0.3296\nINFO     WDM:logger.py:11 Driver [C:\\Users\\Dell\\.wdm\\drivers\\edgedriver\\win64\\137.0.3296.93\\msedgedriver.exe] found in cache\nINFO     utils.driver_manager:driver_manager.py:93 Edge WebDriver created successfully\nINFO     utils.driver_manager:driver_manager.py:110 Driver configured with timeouts - Implicit: 3s, Page load: 10s, Explicit: 8s\n\n----------------------------- Captured stderr call -----------------------------\nINFO - Navigating to blog list: http://localhost:5173/blogs\n\n------------------------------ Captured log call -------------------------------\nINFO     BlogPageNew:blog_page_new.py:90 Navigating to blog list: http://localhost:5173/blogs\n\n--------------------------- Captured stdout teardown ---------------------------\n\n\u2705 Test Suite Complete - 2025-06-23 15:32:18.474258\n\n--------------------------- Captured stderr teardown ---------------------------\nINFO - WebDriver quit successfully\n\n---------------------------- Captured log teardown -----------------------------\nINFO     utils.driver_manager:driver_manager.py:190 WebDriver quit successfully&#34;}]}, &#34;renderCollapsed&#34;: [&#34;passed&#34;], &#34;initialSort&#34;: &#34;result&#34;, &#34;title&#34;: &#34;blog_report_20250623_152219.html&#34;}"></div>
    <script>
      (function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
const { getCollapsedCategory, setCollapsedIds } = require('./storage.js')

class DataManager {
    setManager(data) {
        const collapsedCategories = [...getCollapsedCategory(data.renderCollapsed)]
        const collapsedIds = []
        const tests = Object.values(data.tests).flat().map((test, index) => {
            const collapsed = collapsedCategories.includes(test.result.toLowerCase())
            const id = `test_${index}`
            if (collapsed) {
                collapsedIds.push(id)
            }
            return {
                ...test,
                id,
                collapsed,
            }
        })
        const dataBlob = { ...data, tests }
        this.data = { ...dataBlob }
        this.renderData = { ...dataBlob }
        setCollapsedIds(collapsedIds)
    }

    get allData() {
        return { ...this.data }
    }

    resetRender() {
        this.renderData = { ...this.data }
    }

    setRender(data) {
        this.renderData.tests = [...data]
    }

    toggleCollapsedItem(id) {
        this.renderData.tests = this.renderData.tests.map((test) =>
            test.id === id ? { ...test, collapsed: !test.collapsed } : test,
        )
    }

    set allCollapsed(collapsed) {
        this.renderData = { ...this.renderData, tests: [...this.renderData.tests.map((test) => (
            { ...test, collapsed }
        ))] }
    }

    get testSubset() {
        return [...this.renderData.tests]
    }

    get environment() {
        return this.renderData.environment
    }

    get initialSort() {
        return this.data.initialSort
    }
}

module.exports = {
    manager: new DataManager(),
}

},{"./storage.js":8}],2:[function(require,module,exports){
const mediaViewer = require('./mediaviewer.js')
const templateEnvRow = document.getElementById('template_environment_row')
const templateResult = document.getElementById('template_results-table__tbody')

function htmlToElements(html) {
    const temp = document.createElement('template')
    temp.innerHTML = html
    return temp.content.childNodes
}

const find = (selector, elem) => {
    if (!elem) {
        elem = document
    }
    return elem.querySelector(selector)
}

const findAll = (selector, elem) => {
    if (!elem) {
        elem = document
    }
    return [...elem.querySelectorAll(selector)]
}

const dom = {
    getStaticRow: (key, value) => {
        const envRow = templateEnvRow.content.cloneNode(true)
        const isObj = typeof value === 'object' && value !== null
        const values = isObj ? Object.keys(value).map((k) => `${k}: ${value[k]}`) : null

        const valuesElement = htmlToElements(
            values ? `<ul>${values.map((val) => `<li>${val}</li>`).join('')}<ul>` : `<div>${value}</div>`)[0]
        const td = findAll('td', envRow)
        td[0].textContent = key
        td[1].appendChild(valuesElement)

        return envRow
    },
    getResultTBody: ({ testId, id, log, extras, resultsTableRow, tableHtml, result, collapsed }) => {
        const resultBody = templateResult.content.cloneNode(true)
        resultBody.querySelector('tbody').classList.add(result.toLowerCase())
        resultBody.querySelector('tbody').id = testId
        resultBody.querySelector('.collapsible').dataset.id = id

        resultsTableRow.forEach((html) => {
            const t = document.createElement('template')
            t.innerHTML = html
            resultBody.querySelector('.collapsible').appendChild(t.content)
        })

        if (log) {
            // Wrap lines starting with "E" with span.error to color those lines red
            const wrappedLog = log.replace(/^E.*$/gm, (match) => `<span class="error">${match}</span>`)
            resultBody.querySelector('.log').innerHTML = wrappedLog
        } else {
            resultBody.querySelector('.log').remove()
        }

        if (collapsed) {
            resultBody.querySelector('.collapsible > td')?.classList.add('collapsed')
            resultBody.querySelector('.extras-row').classList.add('hidden')
        } else {
            resultBody.querySelector('.collapsible > td')?.classList.remove('collapsed')
        }

        const media = []
        extras?.forEach(({ name, format_type, content }) => {
            if (['image', 'video'].includes(format_type)) {
                media.push({ path: content, name, format_type })
            }

            if (format_type === 'html') {
                resultBody.querySelector('.extraHTML').insertAdjacentHTML('beforeend', `<div>${content}</div>`)
            }
        })
        mediaViewer.setup(resultBody, media)

        // Add custom html from the pytest_html_results_table_html hook
        tableHtml?.forEach((item) => {
            resultBody.querySelector('td[class="extra"]').insertAdjacentHTML('beforeend', item)
        })

        return resultBody
    },
}

module.exports = {
    dom,
    htmlToElements,
    find,
    findAll,
}

},{"./mediaviewer.js":6}],3:[function(require,module,exports){
const { manager } = require('./datamanager.js')
const { doSort } = require('./sort.js')
const storageModule = require('./storage.js')

const getFilteredSubSet = (filter) =>
    manager.allData.tests.filter(({ result }) => filter.includes(result.toLowerCase()))

const doInitFilter = () => {
    const currentFilter = storageModule.getVisible()
    const filteredSubset = getFilteredSubSet(currentFilter)
    manager.setRender(filteredSubset)
}

const doFilter = (type, show) => {
    if (show) {
        storageModule.showCategory(type)
    } else {
        storageModule.hideCategory(type)
    }

    const currentFilter = storageModule.getVisible()
    const filteredSubset = getFilteredSubSet(currentFilter)
    manager.setRender(filteredSubset)

    const sortColumn = storageModule.getSort()
    doSort(sortColumn, true)
}

module.exports = {
    doFilter,
    doInitFilter,
}

},{"./datamanager.js":1,"./sort.js":7,"./storage.js":8}],4:[function(require,module,exports){
const { redraw, bindEvents, renderStatic } = require('./main.js')
const { doInitFilter } = require('./filter.js')
const { doInitSort } = require('./sort.js')
const { manager } = require('./datamanager.js')
const data = JSON.parse(document.getElementById('data-container').dataset.jsonblob)

function init() {
    manager.setManager(data)
    doInitFilter()
    doInitSort()
    renderStatic()
    redraw()
    bindEvents()
}

init()

},{"./datamanager.js":1,"./filter.js":3,"./main.js":5,"./sort.js":7}],5:[function(require,module,exports){
const { dom, find, findAll } = require('./dom.js')
const { manager } = require('./datamanager.js')
const { doSort } = require('./sort.js')
const { doFilter } = require('./filter.js')
const {
    getVisible,
    getCollapsedIds,
    setCollapsedIds,
    getSort,
    getSortDirection,
    possibleFilters,
} = require('./storage.js')

const removeChildren = (node) => {
    while (node.firstChild) {
        node.removeChild(node.firstChild)
    }
}

const renderStatic = () => {
    const renderEnvironmentTable = () => {
        const environment = manager.environment
        const rows = Object.keys(environment).map((key) => dom.getStaticRow(key, environment[key]))
        const table = document.getElementById('environment')
        removeChildren(table)
        rows.forEach((row) => table.appendChild(row))
    }
    renderEnvironmentTable()
}

const addItemToggleListener = (elem) => {
    elem.addEventListener('click', ({ target }) => {
        const id = target.parentElement.dataset.id
        manager.toggleCollapsedItem(id)

        const collapsedIds = getCollapsedIds()
        if (collapsedIds.includes(id)) {
            const updated = collapsedIds.filter((item) => item !== id)
            setCollapsedIds(updated)
        } else {
            collapsedIds.push(id)
            setCollapsedIds(collapsedIds)
        }
        redraw()
    })
}

const renderContent = (tests) => {
    const sortAttr = getSort(manager.initialSort)
    const sortAsc = JSON.parse(getSortDirection())
    const rows = tests.map(dom.getResultTBody)
    const table = document.getElementById('results-table')
    const tableHeader = document.getElementById('results-table-head')

    const newTable = document.createElement('table')
    newTable.id = 'results-table'

    // remove all sorting classes and set the relevant
    findAll('.sortable', tableHeader).forEach((elem) => elem.classList.remove('asc', 'desc'))
    tableHeader.querySelector(`.sortable[data-column-type="${sortAttr}"]`)?.classList.add(sortAsc ? 'desc' : 'asc')
    newTable.appendChild(tableHeader)

    if (!rows.length) {
        const emptyTable = document.getElementById('template_results-table__body--empty').content.cloneNode(true)
        newTable.appendChild(emptyTable)
    } else {
        rows.forEach((row) => {
            if (!!row) {
                findAll('.collapsible td:not(.col-links', row).forEach(addItemToggleListener)
                find('.logexpander', row).addEventListener('click',
                    (evt) => evt.target.parentNode.classList.toggle('expanded'),
                )
                newTable.appendChild(row)
            }
        })
    }

    table.replaceWith(newTable)
}

const renderDerived = () => {
    const currentFilter = getVisible()
    possibleFilters.forEach((result) => {
        const input = document.querySelector(`input[data-test-result="${result}"]`)
        input.checked = currentFilter.includes(result)
    })
}

const bindEvents = () => {
    const filterColumn = (evt) => {
        const { target: element } = evt
        const { testResult } = element.dataset

        doFilter(testResult, element.checked)
        const collapsedIds = getCollapsedIds()
        const updated = manager.renderData.tests.map((test) => {
            return {
                ...test,
                collapsed: collapsedIds.includes(test.id),
            }
        })
        manager.setRender(updated)
        redraw()
    }

    const header = document.getElementById('environment-header')
    header.addEventListener('click', () => {
        const table = document.getElementById('environment')
        table.classList.toggle('hidden')
        header.classList.toggle('collapsed')
    })

    findAll('input[name="filter_checkbox"]').forEach((elem) => {
        elem.addEventListener('click', filterColumn)
    })

    findAll('.sortable').forEach((elem) => {
        elem.addEventListener('click', (evt) => {
            const { target: element } = evt
            const { columnType } = element.dataset
            doSort(columnType)
            redraw()
        })
    })

    document.getElementById('show_all_details').addEventListener('click', () => {
        manager.allCollapsed = false
        setCollapsedIds([])
        redraw()
    })
    document.getElementById('hide_all_details').addEventListener('click', () => {
        manager.allCollapsed = true
        const allIds = manager.renderData.tests.map((test) => test.id)
        setCollapsedIds(allIds)
        redraw()
    })
}

const redraw = () => {
    const { testSubset } = manager

    renderContent(testSubset)
    renderDerived()
}

module.exports = {
    redraw,
    bindEvents,
    renderStatic,
}

},{"./datamanager.js":1,"./dom.js":2,"./filter.js":3,"./sort.js":7,"./storage.js":8}],6:[function(require,module,exports){
class MediaViewer {
    constructor(assets) {
        this.assets = assets
        this.index = 0
    }

    nextActive() {
        this.index = this.index === this.assets.length - 1 ? 0 : this.index + 1
        return [this.activeFile, this.index]
    }

    prevActive() {
        this.index = this.index === 0 ? this.assets.length - 1 : this.index -1
        return [this.activeFile, this.index]
    }

    get currentIndex() {
        return this.index
    }

    get activeFile() {
        return this.assets[this.index]
    }
}


const setup = (resultBody, assets) => {
    if (!assets.length) {
        resultBody.querySelector('.media').classList.add('hidden')
        return
    }

    const mediaViewer = new MediaViewer(assets)
    const container = resultBody.querySelector('.media-container')
    const leftArrow = resultBody.querySelector('.media-container__nav--left')
    const rightArrow = resultBody.querySelector('.media-container__nav--right')
    const mediaName = resultBody.querySelector('.media__name')
    const counter = resultBody.querySelector('.media__counter')
    const imageEl = resultBody.querySelector('img')
    const sourceEl = resultBody.querySelector('source')
    const videoEl = resultBody.querySelector('video')

    const setImg = (media, index) => {
        if (media?.format_type === 'image') {
            imageEl.src = media.path

            imageEl.classList.remove('hidden')
            videoEl.classList.add('hidden')
        } else if (media?.format_type === 'video') {
            sourceEl.src = media.path

            videoEl.classList.remove('hidden')
            imageEl.classList.add('hidden')
        }

        mediaName.innerText = media?.name
        counter.innerText = `${index + 1} / ${assets.length}`
    }
    setImg(mediaViewer.activeFile, mediaViewer.currentIndex)

    const moveLeft = () => {
        const [media, index] = mediaViewer.prevActive()
        setImg(media, index)
    }
    const doRight = () => {
        const [media, index] = mediaViewer.nextActive()
        setImg(media, index)
    }
    const openImg = () => {
        window.open(mediaViewer.activeFile.path, '_blank')
    }
    if (assets.length === 1) {
        container.classList.add('media-container--fullscreen')
    } else {
        leftArrow.addEventListener('click', moveLeft)
        rightArrow.addEventListener('click', doRight)
    }
    imageEl.addEventListener('click', openImg)
}

module.exports = {
    setup,
}

},{}],7:[function(require,module,exports){
const { manager } = require('./datamanager.js')
const storageModule = require('./storage.js')

const genericSort = (list, key, ascending, customOrder) => {
    let sorted
    if (customOrder) {
        sorted = list.sort((a, b) => {
            const aValue = a.result.toLowerCase()
            const bValue = b.result.toLowerCase()

            const aIndex = customOrder.findIndex((item) => item.toLowerCase() === aValue)
            const bIndex = customOrder.findIndex((item) => item.toLowerCase() === bValue)

            // Compare the indices to determine the sort order
            return aIndex - bIndex
        })
    } else {
        sorted = list.sort((a, b) => a[key] === b[key] ? 0 : a[key] > b[key] ? 1 : -1)
    }

    if (ascending) {
        sorted.reverse()
    }
    return sorted
}

const durationSort = (list, ascending) => {
    const parseDuration = (duration) => {
        if (duration.includes(':')) {
            // If it's in the format "HH:mm:ss"
            const [hours, minutes, seconds] = duration.split(':').map(Number)
            return (hours * 3600 + minutes * 60 + seconds) * 1000
        } else {
            // If it's in the format "nnn ms"
            return parseInt(duration)
        }
    }
    const sorted = list.sort((a, b) => parseDuration(a['duration']) - parseDuration(b['duration']))
    if (ascending) {
        sorted.reverse()
    }
    return sorted
}

const doInitSort = () => {
    const type = storageModule.getSort(manager.initialSort)
    const ascending = storageModule.getSortDirection()
    const list = manager.testSubset
    const initialOrder = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed', 'Skipped', 'Passed']

    storageModule.setSort(type)
    storageModule.setSortDirection(ascending)

    if (type?.toLowerCase() === 'original') {
        manager.setRender(list)
    } else {
        let sortedList
        switch (type) {
        case 'duration':
            sortedList = durationSort(list, ascending)
            break
        case 'result':
            sortedList = genericSort(list, type, ascending, initialOrder)
            break
        default:
            sortedList = genericSort(list, type, ascending)
            break
        }
        manager.setRender(sortedList)
    }
}

const doSort = (type, skipDirection) => {
    const newSortType = storageModule.getSort(manager.initialSort) !== type
    const currentAsc = storageModule.getSortDirection()
    let ascending
    if (skipDirection) {
        ascending = currentAsc
    } else {
        ascending = newSortType ? false : !currentAsc
    }
    storageModule.setSort(type)
    storageModule.setSortDirection(ascending)

    const list = manager.testSubset
    const sortedList = type === 'duration' ? durationSort(list, ascending) : genericSort(list, type, ascending)
    manager.setRender(sortedList)
}

module.exports = {
    doInitSort,
    doSort,
}

},{"./datamanager.js":1,"./storage.js":8}],8:[function(require,module,exports){
const possibleFilters = [
    'passed',
    'skipped',
    'failed',
    'error',
    'xfailed',
    'xpassed',
    'rerun',
]

const getVisible = () => {
    const url = new URL(window.location.href)
    const settings = new URLSearchParams(url.search).get('visible')
    const lower = (item) => {
        const lowerItem = item.toLowerCase()
        if (possibleFilters.includes(lowerItem)) {
            return lowerItem
        }
        return null
    }
    return settings === null ?
        possibleFilters :
        [...new Set(settings?.split(',').map(lower).filter((item) => item))]
}

const hideCategory = (categoryToHide) => {
    const url = new URL(window.location.href)
    const visibleParams = new URLSearchParams(url.search).get('visible')
    const currentVisible = visibleParams ? visibleParams.split(',') : [...possibleFilters]
    const settings = [...new Set(currentVisible)].filter((f) => f !== categoryToHide).join(',')

    url.searchParams.set('visible', settings)
    window.history.pushState({}, null, unescape(url.href))
}

const showCategory = (categoryToShow) => {
    if (typeof window === 'undefined') {
        return
    }
    const url = new URL(window.location.href)
    const currentVisible = new URLSearchParams(url.search).get('visible')?.split(',').filter(Boolean) ||
        [...possibleFilters]
    const settings = [...new Set([categoryToShow, ...currentVisible])]
    const noFilter = possibleFilters.length === settings.length || !settings.length

    noFilter ? url.searchParams.delete('visible') : url.searchParams.set('visible', settings.join(','))
    window.history.pushState({}, null, unescape(url.href))
}

const getSort = (initialSort) => {
    const url = new URL(window.location.href)
    let sort = new URLSearchParams(url.search).get('sort')
    if (!sort) {
        sort = initialSort || 'result'
    }
    return sort
}

const setSort = (type) => {
    const url = new URL(window.location.href)
    url.searchParams.set('sort', type)
    window.history.pushState({}, null, unescape(url.href))
}

const getCollapsedCategory = (renderCollapsed) => {
    let categories
    if (typeof window !== 'undefined') {
        const url = new URL(window.location.href)
        const collapsedItems = new URLSearchParams(url.search).get('collapsed')
        switch (true) {
        case !renderCollapsed && collapsedItems === null:
            categories = ['passed']
            break
        case collapsedItems?.length === 0 || /^["']{2}$/.test(collapsedItems):
            categories = []
            break
        case /^all$/.test(collapsedItems) || collapsedItems === null && /^all$/.test(renderCollapsed):
            categories = [...possibleFilters]
            break
        default:
            categories = collapsedItems?.split(',').map((item) => item.toLowerCase()) || renderCollapsed
            break
        }
    } else {
        categories = []
    }
    return categories
}

const getSortDirection = () => JSON.parse(sessionStorage.getItem('sortAsc')) || false
const setSortDirection = (ascending) => sessionStorage.setItem('sortAsc', ascending)

const getCollapsedIds = () => JSON.parse(sessionStorage.getItem('collapsedIds')) || []
const setCollapsedIds = (list) => sessionStorage.setItem('collapsedIds', JSON.stringify(list))

module.exports = {
    getVisible,
    hideCategory,
    showCategory,
    getCollapsedIds,
    setCollapsedIds,
    getSort,
    setSort,
    getSortDirection,
    setSortDirection,
    getCollapsedCategory,
    possibleFilters,
}

},{}]},{},[4]);
    </script>
  </footer>
</html>